pkg/controller/workload/workload_processor.go

```diff
@@ -501,8 +580,42 @@ func (p *Processor) storeServiceData(serviceName string, waypoint *workloadapi.G
 		newValue.EndpointCount = oldValue.EndpointCount
 	}
 
-	if err = p.bpf.ServiceUpdate(&sk, &newValue); err != nil {
-		log.Errorf("Update Service failed, err:%s", err)
+	// if it is a policy update
+	if newValue.LbPolicy != oldValue.LbPolicy {
+		if newValue.LbPolicy == uint32(workloadapi.LoadBalancing_UNSPECIFIED_MODE) { // from locality loadbalance to random
+			// In locality load balancing mode, the workloads are stored according to the calculated corresponding priorities.
+			// When switching from locality load balancing mode to random, we first update the endpoint map, as at this point,
+			// there might not be any workload with the highest priority, and directly switching the service's LB policy could
+			// lead to unexpected service disruptions.
+			if err = p.updateEndpoint(sk.ServiceId, false); err != nil { // this will change bpf map totally
+				log.Errorf("UpdateEndpoint failed, err:%s", err)
+			}
+			updateValue := bpf.ServiceValue{}
+			if err = p.bpf.ServiceLookup(&sk, &updateValue); err != nil {
+				log.Warnf("Lookup Service failed, err:%s", err)
+			}
+			updateValue.LbPolicy = newValue.LbPolicy
+			if err = p.bpf.ServiceUpdate(&sk, &updateValue); err != nil {
+				log.Errorf("Update Service failed, err:%s", err)
+			}
+			return nil
+		} else if oldValue.LbPolicy == uint32(workloadapi.LoadBalancing_UNSPECIFIED_MODE) { // from random to locality loadbalance
+			// In random mode, the workloads are stored with the highest priority. When switching from random mode to locality
+			// load balancing, we first update the service map to quickly initiate the transition of the strategy. Subsequently,
+			// we update the endpoint map. During this update process, the load balancer may briefly exhibit abnormal random behavior,
+			// after which it will fully transition to the locality load balancing mode.
+			if err = p.bpf.ServiceUpdate(&sk, &newValue); err != nil { // update service first
```

Comment:
ditto EndpointCount not updated?

Reply:
when we changing lb policy, its means that we are in service updating, so the total count number of oldValue and new Value is same.